base/kra/src/main/java/org/dogtagpki/server/kra/rest/base/KeyRequestProcessor.java

```diff
@@ -0,0 +1,1050 @@
+        }
+        if (userPrincipal == null || userPrincipal.getName() == null) {
+            throw new UnauthorizedException("Request approval must be initiated by an agent");
+        }
+        try {
+            Request request = requestRepository.readRequest(id);
+            String realm = request.getRealm();
+            authz.checkRealm(realm, getAuthToken(userPrincipal),
+                    request.getExtDataInString(Request.ATTR_REQUEST_OWNER),
+                    "certServer.kra.requests", "execute");
+            request.setRequestStatus(RequestStatus.REJECTED);
+            requestRepository.updateRequest(request);
+            auditor.log(new SecurityDataRecoveryStateChangeEvent(
+                    userPrincipal.getName(),
+                    ILogger.SUCCESS,
+                    id,
+                    "reject"));
+        }catch (EAuthzAccessDenied e) {
+            throw new UnauthorizedException("Not authorized to reject request", e);
+        } catch (EBaseException e) {
+            auditor.log(new SecurityDataRecoveryStateChangeEvent(
+                    userPrincipal.getName(),
+                    ILogger.FAILURE,
+                    id,
+                    "reject"));
+            throw new PKIException(e.toString(), e);
+        }
+    }
+
+    public void cancel(Principal userPrincipal, RequestId id) {
+        if (id == null) {
+            throw new BadRequestException("Invalid request id.");
+        }
+        if (userPrincipal == null || userPrincipal.getName() == null) {
+            throw new UnauthorizedException("Request approval must be initiated by an agent");
+        }
+        // auth and authz
+        try {
+            Request request = requestRepository.readRequest(id);
+            String realm = request.getRealm();
+            authz.checkRealm(realm, getAuthToken(userPrincipal),
+                    request.getExtDataInString(Request.ATTR_REQUEST_OWNER),
+                    "certServer.kra.requests", "execute");
+            request.setRequestStatus(RequestStatus.CANCELED);
+            requestRepository.updateRequest(request);
+            auditor.log(new SecurityDataRecoveryStateChangeEvent(
+                    userPrincipal.getName(),
+                    ILogger.SUCCESS,
+                    id,
+                    "cancel"));
+        } catch (EAuthzAccessDenied e) {
+            throw new UnauthorizedException("Not authorized to cancel request", e);
+        } catch (EBaseException e) {
+            e.printStackTrace();
+            auditor.log(new SecurityDataRecoveryStateChangeEvent(
+                    userPrincipal.getName(),
+                    ILogger.FAILURE,
+                    id,
+                    "cancel"));
+            throw new PKIException(e.toString(), e);
+        }
+    }
+
+    private KeyRequestResponse generateASymKey(Principal userPrincipal, String baseUrl, AsymKeyGenerationRequest data) {
```

Comment:
IIUC this method is generating an asymmetric key (as opposed to a symmetric key), so the method probably should be called `generateAsymKey()` with lower case `s` to avoid confusions.